keps/sig-node/4622-topologymanager-max-allowable-numa-nodes/README.md

```diff
@@ -0,0 +1,757 @@
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As developer in the AI space, I want to use AI accelerators "super chips" which expose ARM cores with more than 8 NUMA nodes.  
+
+#### Story 2
+
+As user in the high-performance-computing space, I want to enable the sub-NUMA or NUMA-per-socket option of my high-end x86 CPU, which will bring  
+the count of NUMA nodes to exceed 8.  
+
+#### Story 3
+
+As administrator of edge nodes, I want to use power-efficient yet massively parallel ARM chips which expose more 
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+Setting values higher than the current default may cause performance degradation on admission time, because this is the flow on which the Topology Manager operates. Fixing this will require rearchitecting the Topology Manager and it is thus out of scope of this KEP.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+The risk associated with implementing this new proposal is minimal. It pertains only to a distinct policy option within the `TopologyManager` and is safeguarded by the option's inherent security measures, in addition to the default deactivation of the `TopologyManagerPolicyBetaOptions` feature gate.
+
+| Risk                                             | Impact | Mitigation |
+| -------------------------------------------------| -------| ---------- |
+| Set a value lower 8 causes kubelet crash         | High   | the minimum value legal value should be the current hardcoded value(8), If not, we should log it and fail |
+| Set a value too high                             |  Low   | add a log when starting. If possible, we should mark the node Degraded somehow because allocation performance could be significantly slow |
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Users can configure the value of maxAllowableNUMANodes in the TopologyManager when the kubelet starts up, It will be fail and abort if the user set the value is lower than the current hardcoded default (8).
```

Comment:
I don't understand this. Why do we need to error out in this case?
```suggestion
Users can configure the value of maxAllowableNUMANodes in the TopologyManager when the kubelet starts up, It will fail and abort if the user sets the value is lower than the current hardcoded default (8).
```

Reply:
same reasoning as https://github.com/kubernetes/enhancements/pull/4624#discussion_r1629276775

Reply:
Ack